place_tracker/lib/place_map.dart

```diff
@@ -152,16 +151,17 @@ class PlaceMapState extends State<PlaceMap> {
     _showPlacesForSelectedCategory(category);
   }
 
-  void _showPlacesForSelectedCategory(PlaceCategory category) async {
-    await _markedPlaces.forEach((Marker marker, Place place) {
-      mapController.updateMarker(
+  Future<void> _showPlacesForSelectedCategory(PlaceCategory category) async {
+    await Future.forEach(_markedPlaces.keys, (Marker marker) async {
```

Comment:
I think a better way to write this is:

await Future.wait(_markedPlaces.forEach(....).toList());

The way in the code currently will wait for each one to return before proceeding to the next one, and I think what you want is to only proceed once all of these have completed, but one doesn't need to finish before the next.

Reply:
and actually, you don't even need the .toList(). Just:

await Future.wait(_markedPlaces.forEach(....));

Reply:
Ooh, you're right. I hadn't taken the serial vs parallel part into account.

I don't think the snippet in your last comment will work, though, because Future.wait wants an iterable of Futures, and _markedPlaces.forEach is going to return void (though your Dart-fu obviously exceeds mine).

I'll see if I can combine your stuff and my stuff and get something to do the job.

Reply:
I think you want

```dart
Future.wait(_markedPlaces.map((marker, place) => getTheFuture(marker, place)));
```